Add tests for protobuf spans

ryanrhee opened the pull request.

buildbreaker commented on lightstep-spans-proto-lite/src/test/java/com/lightstep/tracer/spans/protolite/ProtobufSpanTest.java:
```diff
@@ -0,0 +1,44 @@
+public final class ProtobufSpanTest {
+
+    @Test
+    public void testTags() {
```
Not fully understanding the test here. What are we testing about the tags?

ryanrhee replied:
i'll make the name more meaningful.

this test checks the case when a tag is added via the span builder, then another tag is added after span creation, before finish(). in this case, the resulting spans should be both of those, and no more.

ryanrhee commented:
whoops, this got merged as a part of https://github.com/lightstep/lightstep-tracer-android3/pull/14

i'll add addressing the comments to this branch
